libs/api/src/services/admin/endpoints.ts

```diff
@@ -574,3 +574,35 @@ export type GetWriteInImageResponse = WriteInImageEntry[] | ErrorsResponse;
  */
 export const GetWriteInImageResponseSchema: z.ZodSchema<GetWriteInImageResponse> =
   z.array(WriteInImageEntrySchema);
+
+/**
+ * Schema for {@link GenerateCardPinRequest}.
+ */
+export const GenerateCardPinRequestSchema = z.never();
+
+/**
+ * @url /admin/pins/generate-card-pin
+ * @method POST
+ */
+export type GenerateCardPinRequest = z.TypeOf<
+  typeof GenerateCardPinRequestSchema
+>;
+
+/**
+ * Schema for {@link GenerateCardPinResponse}.
+ */
+export const GenerateCardPinResponseSchema = z.discriminatedUnion('status', [
+  z.object({
+    status: z.literal('ok'),
+    pin: z.string(),
+  }),
+  ErrorsResponseSchema,
+]);
+
+/**
+ * @url /admin/pins/generate-card-pin
+ * @method POST
+ */
+export type GenerateCardPinResponse = z.TypeOf<
+  typeof GenerateCardPinResponseSchema
+>;
```

Comment:
I haven't used the inferred types from the zod schema because the type values shown by TS when hovering or in error messages loses the type names. See [this TS playground](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAbzgLzgXzgMyhEcDkyEAJvgNwBQFApgB6SxwDGEAdgM7wCiUOU7AJWrtIHagGUmAC2ogAhnAC8KAHQQARgCtqTGAAoEFOHE5yYAV3YAuVQBtgMalDm29+J33wBKADRG4HtDWqnK8cgCeeshqWjr6SDDhYNQ20ZxQwKwA5nq+cCDC7HJZKarpmTle6F6+FGhelDT00PCJyXA8fILCouzUSqoAKknUAPKYADxt1BCYHbxBQiJsfZIy8gB8jQD0AFS7Rrtwa7IKmNCIAAL2rADWcADi1KxOZtQAwqHEAAqZS73UNAqQ7bJoMeAsDjwJ4vZyOT5QH5-HorCTSU4DaLmVjANh6ADa-miGm0ugM-mMpgswWi9kczlc+Agt28fmMxjAmVSKnK2VybOqAs6ixRYhO8j8AF0GlQ9gc4EdLuYoLY4Ns5MQQJltpyONsSrC3gBaJhfI26w5wS4FGBSEhwb6jcSDEFglpwaaPZ6veFfX6sf6ozEqYbJcYTfzTWZew2+xH+wNi9HyChbWX7S1KlVqjVa1g6zLsfXeuHUE1mi0Kq02u3EB1Ol0K0F0cEekZwACCthgMJ9Hz9yOWYiU-gAPogKeyqZYbEyWZR2ezdTZeVkF+y0GP5l1E30dhmq+KzhcENdMvcuz2S28EUiA6K+kDXS33ZDOJ3u73S7eEw+0esFGULEcTxQljGJWIyUMRdpxpFQ6VeRlmVZSdlzKGAMj5WpjHqIUFn4Xd-1OKUZQoYgdFsUJ+jfeAoD-GwvxvAd7yHPcKG2bZFy49kAD0AH4qHIphKLo5gVngFwYEImxL0YuM70IygOO47j+KAA) which compares the displayed type of this form vs the original form. I agree that this would be better if not for this problem:

<img width="952" alt="image" src="https://user-images.githubusercontent.com/1938/196792938-01581bfd-4832-4b5b-b458-804b23742680.png">